x-pack/test/security_solution_cypress/cypress/e2e/explore/hosts/misconfiguration_contextual_flyout.cy.ts

```diff
@@ -0,0 +1,157 @@
+const CSP_INSIGHT_TABLE = getDataTestSubjectSelector(
+  'securitySolutionFlyoutMisconfigurationFindingsTable'
+);
+
+const timestamp = Date.now();
+
+// Create a Date object using the timestamp
+const date = new Date(timestamp);
+
+// Convert the Date object to ISO 8601 format
+const iso8601String = date.toISOString();
+const mockFinding = {
+  '@timestamp': iso8601String,
+  host: { name: 'siem-kibana' },
+  user: { name: 'test' },
+  resource: { id: '1234ABCD', name: `kubelet`, sub_type: 'lower case sub type' },
+  result: { evaluation: 'passed' },
+  rule: {
+    name: 'Upper case rule name',
+    section: 'Upper case section',
+    benchmark: {
+      id: 'cis_k8s',
+      posture_type: 'kspm',
+      name: 'CIS Kubernetes V1.23',
+      version: 'v1.0.0',
+    },
+    type: 'process',
+  },
+  cluster_id: 'Upper case cluster id',
+  data_stream: {
+    dataset: 'cloud_security_posture.findings',
+  },
+};
+
+const createMockFinding = () => {
+  return rootRequest({
+    method: 'POST',
+    url: `${Cypress.env(
+      'ELASTICSEARCH_URL'
+    )}/${CDR_LATEST_NATIVE_MISCONFIGURATIONS_INDEX_PATTERN}/_doc`,
+    body: mockFinding,
+  });
+};
+
+const deleteDataStream = () => {
+  return rootRequest({
+    method: 'DELETE',
+    url: `${Cypress.env(
+      'ELASTICSEARCH_URL'
+    )}/_data_stream/${CDR_LATEST_NATIVE_MISCONFIGURATIONS_INDEX_PATTERN}`,
+  });
+};
+
+describe('Alert Host details expandable flyout', { tags: ['@ess', '@serverless'] }, () => {
+  beforeEach(() => {
+    deleteAlertsAndRules();
+    login();
+    createRule(getNewRule());
+    visit(ALERTS_URL);
+    waitForAlertsToPopulate();
+  });
+
+  // after(() => {
+  //   deleteDataStream();
+  // });
+
+  context('Host name', () => {
+    after(() => {
+      deleteDataStream();
+    });
+
+    it('should not display Misconfiguration preview under Insights Entities when it does not have Misconfiguration Findings', () => {
+      expandFirstAlertHostFlyout();
+
+      cy.log('check if Misconfiguration preview title is not shown');
+      cy.get(CSP_INSIGHT_MISCONFIGURATION_TITLE).should('not.exist');
+    });
+
+    it('should display Misconfiguration preview under Insights Entities when it has Misconfiguration Findings', () => {
+      createMockFinding();
+      cy.reload();
+      expandFirstAlertHostFlyout();
+
+      cy.log('check if Misconfiguration preview title shown');
+      cy.get(CSP_INSIGHT_MISCONFIGURATION_TITLE).should('be.visible');
+    });
+
+    it('should display insight tabs and findings table upon clicking on misconfiguration accordion', () => {
+      createMockFinding();
+      cy.reload();
+      expandFirstAlertHostFlyout();
+      cy.contains('Failed findings', { timeout: 3000 });
+      cy.contains('Misconfigurations').click();
+      cy.get(CSP_INSIGHT_TAB_TITLE).should('be.visible');
+      cy.get(CSP_INSIGHT_TABLE).should('be.visible');
+    });
+  });
+
+  context('User name', () => {
```

Comment:
ideally tests suites should be independant, therefore can ve clean up in `afterEach` (so you create finding in each `it`) and also have separate findings for user and hostname